nixos/modules/services/backup/tsm.nix

```diff
@@ -0,0 +1,301 @@
+      description = ''
+        Create a systemd system service
+        <literal>tsm-backup.service</literal> that starts
+        a backup based on the given servername's stanza.
+        Note that this server's
+        <option>passwdDir</option> will default to
+        <filename>/var/lib/tsm-backup/password</filename>
+        (but may be overridden);
+        also, the service will use
+        <filename>/var/lib/tsm-backup</filename>
+        as <literal>HOME</literal> when calling the client.
+        No backup service will be created if this is
+        set to <literal>null</literal> (the default).
+      '';
+    };
+    service.autoTime = lib.options.mkOption {
+      type = lib.types.nullOr (lib.types.strMatching ".+");
+      default = null;
+      example = "12:00";
+      description = ''
+        The backup service will be invoked
+        automatically at the given date/time,
+        which must be in the format described in
+        <citerefentry><refentrytitle>systemd.time</refentrytitle><manvolnum>5</manvolnum></citerefentry>.
+        The default <literal>null</literal>
+        disables automatic backups.
+      '';
+    };
+  };
+
+  cfg = config.services.tsmClient;
+  cfgSvc = config.services.tsmClient.service;
+
+  assertions = [
+    {
+      assertion = checkIUnique
+        (lib.attrsets.mapAttrsToList (k: v: v.name) cfg.servers);
+      message = ''
+        TSM servernames contain duplicate name
+        (note that case doesn't matter!)
+      '';
+    }
+    {
+      assertion = (cfg.defaultServername!=null)->
+        (lib.attrsets.hasAttr cfg.defaultServername cfg.servers);
+      message = "TSM defaultServername not found in list of servers";
+    }
+    {
+      assertion = (cfgSvc.servername!=null)->
+        (lib.attrsets.hasAttr cfgSvc.servername cfg.servers);
+      message = "TSM service servername not found in list of servers";
+    }
+    {
+      assertion = (cfgSvc.servername!=null)->
+        cfg.servers.${cfgSvc.servername}.genPasswd;
+      message = "TSM service requires automatic password generation";
+    }
+    {
+      assertion = (cfgSvc.autoTime!=null)->(cfgSvc.servername!=null);
+      message = "TSM service timer is set, but its servername is not";
+    }
+  ];
+
+  dsmSysText =
+    let
+      stanzas = lib.attrsets.mapAttrsToList (k: v: v.stanza) cfg.servers;
+    in
+      ''
+        ****  IBM Spectrum Protect (Tivoli Storage Manager)
+        ****  client system-options file "dsm.sys".
+        ****  Do not edit!
+        ****  This file is generated by NixOS configuration.
+
+        ${lib.strings.optionalString (cfg.defaultServername!=null) "defaultserver ${cfg.defaultServername}"}
+
+        ${lib.strings.concatStringsSep "\n" stanzas}
+      '';
+
+in
+
+{
+
+  inherit options;
+
+  config = lib.modules.mkIf cfg.enable {
+    inherit assertions;
+    environment.systemPackages = [ pkgs.tsm-client ];
+    environment.etc."tsm-client/cli.dsm.sys".text = dsmSysText;
```

Comment:
No, probably not. I'm not sure if IBM wants the file in `/etc` at all, since the documentation rather states it to be placed in `DSM_DIR` or in the "installation directory" (https://www.ibm.com/support/knowledgecenter/SSEQVQ_8.1.6/client/t_cfg_crtmodoptunix.html, also for TSM 7: https://www.ibm.com/support/knowledgecenter/SSGSG7_7.1.8/client/t_cfg_crtmodoptunix.html). Do you have any IBM documentation that points to `/etc/dsm`? We can use `/etc/dsm` if that is an established name amongst TSM users, otherwise I would consider `/etc/tsm-client` to be more descriptive. Googling for "/etc/dsm" didn't give me anything conclusive, so I'm not sure about that.